src/core/models.py

```diff
@@ -640,6 +726,16 @@ def node_ip(self) -> str:
         """
         return self.k8s.get_node_ip(self.pod_name)
 
+    @property
+    def directory_id(self) -> str:
+        """Directory ID of the node as saved in `meta.properties`."""
+        return self.relation_data.get("directory-id", "")
```

Comment:
**question:** I'm ignorant, what's this for?

Reply:
This is used for KRaft Quorum manager remove-controller command, and also KRaft leader replica-id. Although it could be always read from `meta.properties` but I thought it's good to save it somewhere.